src/main/solution_space/solution_graph.py

```diff
@@ -188,6 +199,9 @@ def add_code_info_chain(self, code_info_chain: Optional[List[Tuple[Code, CodeInf
             for next_code, next_code_info in code_info_chain[1:]:
                 next_vertex = self.find_or_create_vertex(next_code, next_code_info)
                 prev_vertex.add_child(next_vertex)
+                next_anon_tree = next_vertex.serialized_code.get_last_anon_tree()
+                if next_anon_tree:
+                    prev_vertex.serialized_code.get_last_anon_tree().add_next_anon_tree(next_anon_tree)
```

Comment:
Sorry, but now I'm not sure it works :(
Maybe we should have a variable 
`next_anon_tree = next_vertex.serialized_code.find_anon_tree(next_code.anon_tree)`?